### PR TITLE
Accept `Sensitive` mysqld_exporter `cnf_password`

### PR DESCRIPTION
**Pull request by alexjfisher**
Also convert template to `epp`.

### CODE REVIEW

**Comment by alexjfisher** [comment]
Added extra commit.
```
Move mysqld_exporter `cnf_*` defaults into class
    
    Group them together and use puppet-strings tags to document them.
    Previously, it was inconsistent with `cnf_socket` default in the class,
    but all others in hiera.  This also meant it was not possible to group
    them together in the parameter list without getting a puppet-lint
    warning about optional parameters coming before required ones.
```

File: manifests/mysqld_exporter.pp [hunk 1]
```diff
@@ -104,6 +94,14 @@
   String $package_name,
   String $user,
   String $version,
+
+  Stdlib::Absolutepath              $cnf_config_path = '/etc/.my.cnf',
+  Stdlib::Host                      $cnf_host        = localhost,
+  Stdlib::Port                      $cnf_port        = 3306,
+  String[1]                         $cnf_user        = login,
+  Variant[Sensitive[String],String] $cnf_password    = 'password',
```
Comment (bastelfreak) [inline manifests/mysqld_exporter.pp:102]: can you update this?
```suggestion
  Variant[Sensitive[String[1]],String[1]] $cnf_password    = 'password',
```

File: templates/my.cnf.epp [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<%- |
+  String $cnf_user,
+  Variant[Sensitive[String],String] $cnf_password,
+  Stdlib::Port $cnf_port,
+  Stdlib::Host $cnf_host,
+  Optional[Stdlib::Absolutepath] $cnf_socket = undef,
+| -%>
```
Comment (bastelfreak) [inline templates/my.cnf.epp:7]: can you add a comment like `# THIS FILE IS MANAGED BY PUPPET`?

Comment (alexjfisher) [reply templates/my.cnf.epp:7]: sure

Comment (alexjfisher) [reply templates/my.cnf.epp:7]: Done.

File: manifests/mysqld_exporter.pp (hunk 1 above)
Comment (alexjfisher) [reply manifests/mysqld_exporter.pp:102]: An empty password is probably valid though?

**Pull request merged**